### PR TITLE
Laboratory work #4, Yudakova Olesya - 19FPL2

### PR DESCRIPTION
**Pull request by yudakovaolesya**

### CODE REVIEW

File: shapes/circle.py [hunk 1]
```diff
@@ -3,32 +3,38 @@
...
+        super().__init__(uid)
+        self.radius = radius
 
     def get_area(self):
         """
         Returns the area of a circle
         :return int: the area of a circle
         """
-        pass
+        area = pi * (self.radius ** 2)
```
Comment (dmitry-uraev) [inline shapes/circle.py:23]: return the result of this expression directly

File: shapes/circle.py [hunk 2]
```diff
@@ -3,32 +3,38 @@
...
+        return area
 
     def get_perimeter(self):
         """
         Returns the perimeter of a circle
         :return int: the perimeter of a circle
         """
-        pass
+        perimeter = 2 * pi * self.radius
```
Comment (dmitry-uraev) [inline shapes/circle.py:31]: same

File: shapes/circle.py [hunk 3]
```diff
@@ -3,32 +3,38 @@
...
+        return perimeter
 
     def get_diameter(self):
         """
         Returns the diameter of a circle
         :return int: the diameter of a circle
         """
-        pass
+        diameter = 2 * self.radius
```
Comment (dmitry-uraev) [inline shapes/circle.py:39]: same

File: shapes/parallelogram.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        self.width = width
+        self.length = length
+        self.acute_angle = acute_angle
+
+    def get_area(self):
+        """
+        Returns the area of a parallelogram
+        :return int: the area of a parallelogram
+        """
+        area = self.length * self.width * sin(self.acute_angle)
```
Comment (dmitry-uraev) [inline shapes/parallelogram.py:25]: same and so on)

File: shapes/shape.py [hunk 5]
```diff
@@ -12,24 +12,22 @@ class Shape(ABC):
...
         """
-        pass
+        return self.uid
 
     @abstractmethod
     def get_area(self):
         """
         Returns the area of a shape
         """
-        pass
```
Comment (dmitry-uraev) [inline shapes/shape.py:28]: indicate to your user that this method is abstract not only by decorator, but internal method raise

File: shapes/shape.py [hunk 6]
```diff
@@ -12,24 +12,22 @@ class Shape(ABC):
...
 
     @abstractmethod
     def get_perimeter(self):
         """
         Returns the perimeter of a shape
         """
-        pass
```
Comment (dmitry-uraev) [inline shapes/shape.py:35]: same

**Review by dmitry-uraev — COMMENTED** [review]
8

File: shapes/circle.py (hunk 1 above)
Comment (yudakovaolesya) [reply shapes/circle.py]: done

File: shapes/circle.py (hunk 2 above)
Comment (yudakovaolesya) [reply shapes/circle.py]: done

File: shapes/circle.py (hunk 3 above)
Comment (yudakovaolesya) [reply shapes/circle.py]: done

File: shapes/parallelogram.py (hunk 4 above)
Comment (yudakovaolesya) [reply shapes/parallelogram.py]: done

File: shapes/shape.py (hunk 5 above)
Comment (yudakovaolesya) [reply shapes/shape.py:28]: done

File: shapes/shape.py (hunk 6 above)
Comment (yudakovaolesya) [reply shapes/shape.py:35]: done

**Review by dmitry-uraev — COMMENTED** [review]
8

**Pull request closed without merging**